AK/StringConversions.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2022, David Tuin <davidot@serenityos.org>
+ * Copyright (c) 2025, Tim Flynn <trflynn89@ladybird.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/StringConversions.h>
+#include <AK/StringView.h>
+
+#include <fast_float/fast_float.h>
+
+namespace AK {
+
+#define ENUMERATE_ARITHMETIC_TYPES                  \
+    __ENUMERATE_ARITHMETIC_TYPE(i8)                 \
+    __ENUMERATE_ARITHMETIC_TYPE(i16)                \
+    __ENUMERATE_ARITHMETIC_TYPE(i32)                \
+    __ENUMERATE_ARITHMETIC_TYPE(long)               \
```

Comment:
This will break if i32 is long, right? Should the types in this X macro all be primitive types instead of AK/Types types?

Reply:
That's exactly what I had at first but it didn't work on macOS 🤷 

I remember this being a pain in the past, so what I've done here is just use the same specializations that we were already using in StringUtils.cpp.

Reply:
Huh ... Well I mean if we know it works then whatever. But it still stands out

Reply:
For reference: https://github.com/SerenityOS/serenity/commit/accf4b338dee50dcacc7caf5dc46e0abc5dffd89